### ISSUE TITLE
Error parsing intermediate spec file for all openSUSE spec files

### ISSUE THREAD

**Issue by jayvdb** [nphilipp/rpmspectool#13]
I am getting the following

```console
> rpmspectool --debug list *.spec
DEBUG:root:args: Namespace(cmd='list', debug=True, define=[], patch=None, patches=False, source=None, sources=False, specfile=<_io.BufferedReader name='python-djangorestframework.spec'>, verbose=False)
DEBUG:root:Created temporary directory '/tmp/rpmspectool_trrh54g8'
DEBUG:root:eval_specfile()
DEBUG:root:writing parsed file '/tmp/rpmspectool_trrh54g8/rpmspectool-python-djangorestframework.spec'
Error parsing intermediate spec file '/tmp/rpmspectool_trrh54g8/rpmspectool-python-djangorestframework.spec'.
> rpmbuild --version
RPM version 4.14.2.1
```

This happens when trying to use it on every spec file in https://build.opensuse.org/project/show/devel:languages:python , and  a few other openSUSE package .spec .

**Comment by nphilipp** [comment]
Can `rpm` parse the supplied spec file? I'm asking because here it complains about an undefined macro `%python_subpackages`. Please run the following command and check its output:

```
rpm --specfile python-djangorestframework.spec
```

If you get the same error, you probably need to install the `python-rpm-macros` package.

**Comment by jayvdb** [comment]
Works fine for me.  I do have python-rpm-macros installed locally.

```console
> rpm --specfile python-djangorestframework.spec
python-djangorestframework-3.9.2-0.noarch
python2-djangorestframework-3.9.2-0.noarch
python3-djangorestframework-3.9.2-0.noarch
```

**Comment by nphilipp** [comment]
Hmm, must be something else then. When you run `rpmspectool --debug ...` like above, it tells you the name of the intermediate spec file it writes and tries to parse (and doesn't remove it afterwards). Can you run `rpm --specfile ...` on it, too? And attach the file here? Thanks!

**Comment by nphilipp** [comment]
@jayvdb ping?

**Comment by jayvdb** [comment]
btw, you havent create a tag for the latest release on PyPI.

I uploaded a project
https://build.opensuse.org/package/view_file/home:jayvdb:py-new/python-rpmspectool/python-rpmspectool.spec?expand=1
```spec
#
# spec file for package python-rpmspectool
#
# Copyright (c) 2019 SUSE LINUX GmbH, Nuernberg, Germany.
#
# All modifications and additions to the file contributed by third parties
# remain the property of their copyright owners, unless otherwise agreed
# upon. The license for this file, and modifications and additions to the
# file, is the same license as for the pristine package itself (unless the
# license for the pristine package is not an Open Source License, in which
# case the license is the MIT License). An "Open Source License" is a
# license that conforms to the Open Source Definition (Version 1.9)
# published by the Open Source Initiative.

# Please submit bugfixes or comments via http://bugs.opensuse.org/


%{?!python_module:%define python_module() python-%{**} python3-%{**}}
%define skip_python2 1
Name:           python-rpmspectool
Version:        1.99.5
Release:        0
License:        GPL-3.0
Summary:        Download sources and patches in RPM spec files
Url:            https://github.com/nphilipp/rpmspectool
Group:          Development/Languages/Python
Source:         https://github.com/nphilipp/rpmspectool/archive/rpmspectool-%{version}.tar.gz#/rpmspectool-%{version}.tar.gz
Patch0:         merged_9476fb1e.patch
BuildRequires:  python-rpm-macros
BuildRequires:  %{python_module setuptools}
BuildRequires:  %{python_module pytest}
# SECTION test requirements
BuildRequires:  %{python_module pycurl}
# /SECTION
BuildRequires:  fdupes
Requires:       python-pycurl
BuildArch:      noarch

%python_subpackages

%description
The rpmspectool utility lets people dealing with RPM packages list
and download sources and patches in RPM spec files.

%prep
%setup -q -n rpmspectool-rpmspectool-%{version}
%patch0 -p1

%build
%python_build

%install
%python_install
%python_expand %fdupes %{buildroot}%{$python_sitelib}

#%%check
#%%pytest

%files %{python_files}
%doc README.md
%license COPYING
%python3_only %{_bindir}/rpmspectool
%{python_sitelib}/*

%changelog
```
Here is the generated .spec

```spec
%undefine _topdir
%define _topdir /home/jayvdb/rpmbuild

%undefine _sourcedir
%define _sourcedir /home/jayvdb/rpmbuild/SOURCES

%undefine _builddir
%define _builddir /home/jayvdb/rpmbuild/BUILD

%undefine _srcrpmdir
%define _srcrpmdir /home/jayvdb/rpmbuild/SRPMS

%undefine _rpmdir
%define _rpmdir /home/jayvdb/rpmbuild/RPMS


#
# spec file for package python-rpmspectool
#
# Copyright (c) 2019 SUSE LINUX GmbH, Nuernberg, Germany.
#
# All modifications and additions to the file contributed by third parties
# remain the property of their copyright owners, unless otherwise agreed
# upon. The license for this file, and modifications and additions to the
# file, is the same license as for the pristine package itself (unless the
# license for the pristine package is not an Open Source License, in which
# case the license is the MIT License). An "Open Source License" is a
# license that conforms to the Open Source Definition (Version 1.9)
# published by the Open Source Initiative.

# Please submit bugfixes or comments via http://bugs.opensuse.org/


%{?!python_module:%define python_module() python-%{**} python3-%{**}}
%define skip_python2 1
Name:           python-rpmspectool
Version:        1.99.5
Release:        0
License:        GPL-3.0
Summary:        Download sources and patches in RPM spec files
Url:            https://github.com/nphilipp/rpmspectool
Group:          Development/Languages/Python
Source:         https://github.com/nphilipp/rpmspectool/archive/rpmspectool-%{version}.tar.gz#/rpmspectool-%{version}.tar.gz
Patch0:         merged_9476fb1e.patch
BuildRequires:  python-rpm-macros
BuildRequires:  %{python_module setuptools}
BuildRequires:  %{python_module pytest}
# SECTION test requirements
BuildRequires:  %{python_module pycurl}
# /SECTION
BuildRequires:  fdupes
Requires:       python-pycurl

%python_subpackages

%description
%prep
cat << EOF
#
# spec file for package python-rpmspectool
#
# Copyright (c) 2019 SUSE LINUX GmbH, Nuernberg, Germany.
#
# All modifications and additions to the file contributed by third parties
# remain the property of their copyright owners, unless otherwise agreed
# upon. The license for this file, and modifications and additions to the
# file, is the same license as for the pristine package itself (unless the
# license for the pristine package is not an Open Source License, in which
# case the license is the MIT License). An "Open Source License" is a
# license that conforms to the Open Source Definition (Version 1.9)
# published by the Open Source Initiative.

# Please submit bugfixes or comments via http://bugs.opensuse.org/


%{?!python_module:%define python_module() python-%{**} python3-%{**}}
%define skip_python2 1
Name:           python-rpmspectool
Version:        1.99.5
Release:        0
License:        GPL-3.0
Summary:        Download sources and patches in RPM spec files
Url:            https://github.com/nphilipp/rpmspectool
Group:          Development/Languages/Python
Source:         https://github.com/nphilipp/rpmspectool/archive/rpmspectool-%{version}.tar.gz#/rpmspectool-%{version}.tar.gz
Patch0:         merged_9476fb1e.patch
BuildRequires:  python-rpm-macros
BuildRequires:  %{python_module setuptools}
BuildRequires:  %{python_module pytest}
# SECTION test requirements
BuildRequires:  %{python_module pycurl}
# /SECTION
BuildRequires:  fdupes
Requires:       python-pycurl

%python_subpackages


SrcDir: %{_sourcedir}
EOF
```

My guess is the cause is `%python_subpackages` and all the voodoo happening in https://github.com/openSUSE/python-rpm-macros/blob/master/macros.lua

**Comment by nphilipp** [comment]
Ugh. Looks like I'll have to hardcode this macro to be undefined before attempting to parse things.

**Comment by jayvdb** [comment]
That macro effectively rewrites the .spec, changing many things, so I suspect new problems if the macro isnt executed, but I havent dug into your code so I dont know what is needed out of the parse phase.

**Comment by nphilipp** [comment]
This was fixed long ago in 4980f811c728207a491e1ad1722b99d3250b93f9 (which ended up in version 1.99.7).
